src/js/esse/schemaUtils.js

```diff
@@ -0,0 +1,43 @@
+import keyBy from "lodash/keyBy";
+
+export function mapObjectDeep(object, mapValue) {
+    if (typeof object !== "object") {
+        return object;
+    }
+
+    if (Array.isArray(object)) {
+        return object.map(
+            (innerValue) => mapValue(innerValue) || mapObjectDeep(innerValue, mapValue),
+        );
+    }
+
+    const entries = Object.entries(object).map(([key, value]) => {
+        return [key, mapValue(value) || mapObjectDeep(value, mapValue)];
+    });
+
+    return Object.fromEntries(entries);
+}
+
+export function makeFlatSchemaId(schemaId) {
+    return schemaId.replace(/[^a-zA-Z]/g, "-");
```

Comment:
Should we use `:` for uniqueness, b/c `-` are used in filenames (basenames) already